### DISCUSSION TITLE
I don't completely understand external recipes

### DISCUSSION THREAD

**Discussion by speendo** [TandoorRecipes/recipes#1557] [Q&A]
First of all thank you for this great project!

I try to import my mother's cookbooks and recipes which she stored as a pdf, however, I don't really understand what is happening.

It seems I was able to connect Tandoor with my Nextcloud. However, although there are two files in the directory, only one is discovered, see here:

![image](https://user-images.githubusercontent.com/1668429/154753255-f6830f48-af6a-44a6-b977-8682e7be71af.png)

Also, after importing the file I don't find the place where I can display the recipe. The doc says it's under `Manage Data >> Discovered recipes` but I cannot find the menu entry `Manage Data`. Forgive me, if this is a stupid question...

**Comment by vabene1111** [comment]
the "entdecken liste" you are showing is "discovered recipes". There you will need to hit import, after that recipes show just as all recipes on the main search page. 

But that only one is imported is strange, can you share the sync log ?

**Reply by speendo** [reply]
Thanks for looking at it, @vabene1111. The other strange thing is that the imported recipe doesn't show up in the list.

Here is the sync log:
![grafik](https://user-images.githubusercontent.com/1668429/154803012-21c0ecbe-9092-416a-8744-cc061d46a30a.png)

Here you can see the recipes from the backend
![grafik](https://user-images.githubusercontent.com/1668429/154803082-6ecfe93d-8c18-49e9-ab1f-38c05c31a1d2.png)

And here you have the frontend
![grafik](https://user-images.githubusercontent.com/1668429/154803114-ef8fe870-0d0f-4003-88ee-b87ed97cc4c4.png)

**Comment by vabene1111** [comment]
do you have the filter for internal set ?

**Comment by speendo** [comment]
Good idea, but no, it's not on.

**Comment by vabene1111** [comment]
that is very interesting. Can you set `DEBUG=1` in you environment config and run the sync again ? If that does not show anything i might need to add custom debug code to see why this is not working

**Reply by smilerz** [reply]
is #1565 related?

**Comment by speendo** [comment]
Thanks again for your help. So I set `DEBUG=1`, deleted the imported recipe in the Django administration backend and recreated the sync. Again it found just one of the two pdfs which I then imported:
![grafik](https://user-images.githubusercontent.com/1668429/155019816-bdacc59b-7160-4992-be82-fa12a89db33d.png)

Again, it did not show up in the recipe list.

I found a Debug log on the `System` page, however, it doesn't seem to provide any relevant information, or does it?:

```
Gunicoren Media: False
Sqlite: False
Debug: True

SERVER_PROTOCOL:HTTP/1.0
REMOTE_ADDR:192.168.XXX.XXX
SERVER_PORT:8080

HTTP_HOST:rezepte.XXX.XXX
HTTP_CONNECTION:close
HTTP_X_FORWARDED_SCHEME:https
HTTP_X_FORWARDED_PROTO:https
HTTP_X_FORWARDED_FOR:XX.XX.XX.XX
HTTP_X_REAL_IP:XX.XX.XX.XX
HTTP_USER_AGENT:Mozilla/5.0 (X11; Linux x86_64; rv:93.0) Gecko/20100101 Firefox/93.0
HTTP_ACCEPT:text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8
HTTP_ACCEPT_LANGUAGE:de-AT,de;q=0.8,en-US;q=0.5,en;q=0.3
HTTP_ACCEPT_ENCODING:gzip, deflate, br
HTTP_UPGRADE_INSECURE_REQUESTS:1
HTTP_DNT:1
HTTP_SEC_FETCH_DEST:empty
HTTP_SEC_FETCH_MODE:same-origin
HTTP_SEC_FETCH_SITE:same-origin
HTTP_CACHE_CONTROL:max-age=0
HTTP_COOKIE:csrftoken=XXXXXXX; sessionid=XXXXX

wsgi.errors:<gunicorn.http.wsgi.WSGIErrorsWrapper object at 0x7efeb8971d60>
wsgi.version:(1, 0)
wsgi.multithread:False
wsgi.multiprocess:False
wsgi.run_once:False
wsgi.file_wrapper:
wsgi.input_terminated:True
wsgi.input:<gunicorn.http.body.Body object at 0x7efeb8971730>
wsgi.url_scheme:http

    
```

If this does help in any way, I can provide you (user) access for my instance and also for the nextcloud I want to sync.

**Comment by vabene1111** [comment]
ah sorry, i did not specify which logs i need. I need the gunicorn logs, in a docker compose setup you will find them using `docker-compose logs` if not docker compose its likely systemclt 

if those logs dont hold anything interesting i will need to add some debug logs to the code of the next release so we can look further into this issue.

**Comment by speendo** [comment]
Ah, I see. I even looked at those but didn't see anything very interesting. But the logs were also cluttered with nginx-messages.

When I am back home, I will look into the container to see if I can extract the gunicorn logs alone.

**Reply by vabene1111** [reply]
with docker compose you can just use `| grep web_recipes` to filter for only those messages, but no hurry

**Comment by speendo** [comment]
this should be the log:
https://nextcloud.yeara.net/s/kawoztJJxQfBLdX

**Reply by speendo** [reply]
(please tell me when you downloaded it, I'd like to remove the share link then. Probably it's paranoia but there might be interesting information on my server config that is not really necessary for others to know.)

**Reply by vabene1111** [reply]
i have reviewed it, sadly no errors that can help me debug. if you have anything confidential you can always email me with info@tandoor.dev 

i will need to add some debug code to the next version to log potential issues and see if that helps us find out why this particular recipe is not working, to be honest its been a very long time since i worked on the external recipes so there might be things that got broken over time.

**Reply by speendo** [reply]
I don't know if this information deserves the attribute "confidential". But a healthy portion of paranoia... :) anyway.

Thanks for looking at it, it's not crucial for the working of my tandoor instance. Still, the feature would be handy, as it might be a possibility to bring my mother's self collected recipe books to my database and also encourage her to participate...

Please let me know if I can provide any additional information. Would it help if I send you the pdf that wasn't imported? Or would you like to have access to the folder I try to import?

**Reply by vabene1111** [reply]
i just found and fixed the issue, i think its related to an API change in nextcloud or the webdav client library i am using. This should work after the next update altough external recipes simple have not received enough love over the past year, really need to work on that agian.

**Reply by speendo** [reply]
Great that you found it! I look forward to the next release :)

**Comment by speendo** [comment]
Thank you!
I can confirm that it works now!
